src/code/z_map_exp.c

```diff
@@ -1,20 +1,23 @@
 #include "global.h"
 #include "overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope.h"
 
-s16 sMinimapInitPosX = 0;
-s16 sMinimapInitPosZ = 0;
-s16 sMinimapInitDir = 0;
-
-s32 sDungeonAndBossSceneIds[] = {
-    SCENE_MITURIN,    SCENE_HAKUGIN,    SCENE_SEA,    SCENE_INISIE_N,  SCENE_INISIE_R,
-    SCENE_MITURIN_BS, SCENE_HAKUGIN_BS, SCENE_SEA_BS, SCENE_INISIE_BS,
-};
+s16 sPlayerInitPosX = 0;
+s16 sPlayerInitPosZ = 0;
+s16 sPlayerInitDir = 0;
 
 /**
- * If the current scene is a dungeon or boss scene, this function returns an index
- * indicating which one it is. Otherwise, it returns -1.
+ * Gets the mapIndex for the current dungeon or boss scene
+ * @return the current scene's mapIndex, or -1 if it isn't a dungeon or boss scene.
  */
 s32 Map_GetDungeonOrBossAreaIndex(PlayState* play) {
+    static s32 sDungeonAndBossSceneIds[] = {
+        SCENE_MITURIN,  // DUNGEON_INDEX_WOODFALL_TEMPLE
+        SCENE_HAKUGIN,  // DUNGEON_INDEX_SNOWHEAD_TEMPLE
+        SCENE_SEA,      // DUNGEON_INDEX_GREAT_BAY_TEMPLE
+        SCENE_INISIE_N, // DUNGEON_INDEX_STONE_TOWER_TEMPLE
+        SCENE_INISIE_R, // DUNGEON_INDEX_STONE_TOWER_TEMPLE
```

Comment:
```suggestion
        SCENE_INISIE_N, // DUNGEON_INDEX_STONE_TOWER_TEMPLE
        SCENE_INISIE_R,
```

Reply:
The comment is accurate, and imo it should be left in because this part of the system is a mess. 

When `play->sceneId == SCENE_INISIE_R`, the function will return `DUNGEON_INDEX_STONE_TOWER_TEMPLE`; This is because Play_GetOriginalSceneId will convert SCENE_INISIE_R` -> `SCENE_INISIE_N` first before checking if the scene is in the `sDungeonAndBossSceneIds` list.